custom/icds_reports/views.py

```diff
@@ -1949,3 +1949,33 @@ def get_type_code(self, data_type):
             "awc": 3
         }
         return type_map[data_type]
+
+
+@location_safe
+@method_decorator([api_auth, toggles.mwcd_indicators.required_decorator()], name='dispatch')
+class MWCDDataView(View):
+
+    def message(self, message_name):
+        error_messages = {
+            "unknown_error": "Unknown Error occured",
+            "no_data": "Data does not exists"
+        }
+
+        return error_messages[message_name]
+
+    def get(self, request, *args, **kwargs):
+
+        try:
+            data = get_mwcd_indicator_api_data()
+            response = {'isSuccess': True,
+                        'message': 'Data Sent Successfully',
+                        'Result': {
+                            'response': data
+                        }}
+            return JsonResponse(response)
+        except MWCDReportView.DoesNotExist:
```

Comment:
When would this be raised?

Reply:
I do pull some data hereby filtering based on the month. I want to catch this in case the data does not exist for any reason for a corner case say for initial 2-3 days of the month or for any.

Reply:
Won't that just return an empty queryset not `DoesNotExist`. I think that only is returned by `model.objects.get` which you do not use. A filter with no results should not raise an exception. If you want to return something special for that (rather than just empty data which may also be fine), you will have to look for empty data rather than an exception

Reply:
yeah. I changed the code. the client actually does not care about seeing specific errors just want to see when there is an error. So updated the code accordingly